packages/consumption/src/modules/notifications/itemProcessors/attributeDeleted/OwnSharedAttributeDeletedByOwnerNotificationItemProcessor.ts

```diff
@@ -0,0 +1,82 @@
+        notification: LocalNotification
+    ): Promise<ValidationResult> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+
+        if (typeof attribute === "undefined") {
+            return ValidationResult.success();
+        }
+
+        if (!attribute.isPeerSharedAttribute()) {
+            return ValidationResult.error(CoreErrors.attributes.isNotPeerSharedAttribute(notificationItem.attributeId));
+        }
+
+        if (!notification.peer.equals(attribute.shareInfo.peer)) {
+            return ValidationResult.error(CoreErrors.attributes.senderIsNotPeerOfSharedAttribute(notification.peer, notificationItem.attributeId));
+        }
+
+        return ValidationResult.success();
+    }
+
+    public override async process(
+        notificationItem: OwnSharedAttributeDeletedByOwnerNotificationItem,
+        _notification: LocalNotification
+    ): Promise<OwnSharedAttributeDeletedByOwnerEvent | void> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+        if (typeof attribute === "undefined") return;
+
+        const deletionInfo = LocalAttributeDeletionInfo.from({
+            deletionStatus: DeletionStatus.DeletedByOwner,
+            deletionDate: CoreDate.utc()
+        });
+
+        let updatedAttribute = attribute;
+        if (typeof attribute.deletionInfo === "undefined") {
+            attribute.setDeletionInfo(deletionInfo, this.accountController.identity.address);
+            updatedAttribute = await this.consumptionController.attributes.updateAttributeUnsafe(attribute);
+        }
+
+        const predecessors = await this.consumptionController.attributes.getPredecessorsOfAttribute(attribute.id);
+        for (const predecessor of predecessors) {
+            if (typeof predecessor.deletionInfo === "undefined") {
+                predecessor.setDeletionInfo(deletionInfo, this.accountController.identity.address);
+                await this.consumptionController.attributes.updateAttributeUnsafe(predecessor);
+            }
+        }
+
+        return new OwnSharedAttributeDeletedByOwnerEvent(this.currentIdentityAddress.toString(), updatedAttribute);
+    }
+
+    public override async rollback(notificationItem: OwnSharedAttributeDeletedByOwnerNotificationItem, _notification: LocalNotification): Promise<void> {
+        const attribute = await this.consumptionController.attributes.getLocalAttribute(notificationItem.attributeId);
+        if (typeof attribute === "undefined") return;
+
+        attribute.deletionInfo = undefined;
+        await this.consumptionController.attributes.updateAttributeUnsafe(attribute);
+
+        const predecessors = await this.consumptionController.attributes.getPredecessorsOfAttribute(attribute.id);
+        for (const predecessor of predecessors) {
+            predecessor.deletionInfo = undefined;
```

Comment:
Is `DeletedByOwner` -> `undefined` the only legal state transition?

Reply:
Yes, peer shared Attributes may only have the DeletionStatus `ToBeDeleted` (if an according Request was accepted) or `DeletedByOwner`. `DeletedByPeer` and `ToBeDeletedByPeer` are only allowed for own shared Attributes. If the Attribute had the DeletionStatus `ToBeDeleted`, processing this Notification won't change its status (line 53 and 60). Hence, the peer shared Attribute will only be set to the DeletionStatus `DeletedByOwner`, if the DeletionStatus was undefined before. Now, calling the rollback function, we know that we only need to reset it to undefined.